### PR TITLE
.Net: Assistant Experimental Feature Update

### PR DESCRIPTION
**Pull request by crickman**
### Motivation and Context

Refine patterns related to experimental assistants feature.  Most relevantly, addressing a functional bug where all assistant functions (`AskAsync`) have the exact same generic description:

```
"Provide user message to assistant and retrieve the assistant response."
```

The KernelFunction description should capture the assistant description.

### Description

This change fixes several aspects for using assistants as kernel functions/plugin.
- Include assistant specific description to `KernelFunction` definition
- Return entire `AssistantResponse` as function result when processing assistant tool-result.
- Incorporate yaml template into builder pattern (allows for description override)
- Tune test prompts/metadata

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

File: dotnet/samples/KernelSyntaxExamples/Example70_Assistant.cs [hunk 1]
```diff
@@ -120,13 +120,10 @@ private static async Task RunAsFunctionAsync()
...
         var assistant =
-            await AssistantBuilder.FromDefinitionAsync(
-                TestConfiguration.OpenAI.ApiKey,
-                OpenAIFunctionEnabledModel, // Must be a function-enabled model.
-                EmbeddedResource.Read("Assistants.ParrotAssistant.yaml")); // Defined under ./Resources/Assistants
-
-        // Import assistant as a KernelFunction
-        var plugin = KernelPluginFactory.CreateFromObject(assistant, assistant.Id);
+            await new AssistantBuilder()
+                .AddOpenAIChatCompletion(OpenAIFunctionEnabledModel, TestConfiguration.OpenAI.ApiKey)
```
Comment (SergeyMenshykh) [inline dotnet/samples/KernelSyntaxExamples/Example70_Assistant.cs:124]: Should the builder method name follow the same "With*" pattern as the other methods in `AssistantBuilder` and `KernelBuilder`?

Comment (crickman) [reply dotnet/samples/KernelSyntaxExamples/Example70_Assistant.cs:124]: I was wondering the same.  Matthew picked the name, so I've just been rolling with it...let's take that alignment now. Yes.

Comment (crickman) [reply dotnet/samples/KernelSyntaxExamples/Example70_Assistant.cs:124]: Done

File: dotnet/samples/KernelSyntaxExamples/Example71_AssistantDelegation.cs [hunk 2]
```diff
@@ -36,26 +36,25 @@ public static async Task RunAsync()
...
         var menuAssistant =
-            await AssistantBuilder.FromDefinitionAsync(
-                TestConfiguration.OpenAI.ApiKey,
-                model: OpenAIFunctionEnabledModel,
-                template: EmbeddedResource.Read("Assistants.ToolAssistant.yaml"),
-                new[] { plugin });
+            await new AssistantBuilder()
+                .WithOpenAIChatCompletion(OpenAIFunctionEnabledModel, TestConfiguration.OpenAI.ApiKey)
+                .FromTemplate(EmbeddedResource.Read("Assistants.ToolAssistant.yaml"))
+                .WithDescription("Answer questions about the menu use the tool.")
```
Comment (glahaye) [inline dotnet/samples/KernelSyntaxExamples/Example71_AssistantDelegation.cs:42]: A couple of typos here.  Maybe you meant something like:

```suggestion
                .WithDescription("Answer questions about how the menu uses the tool.")
```

**Review by markwallace-microsoft — APPROVED** [review]

**Review by glahaye — APPROVED** [review]

**Pull request merged**